### PR TITLE
sql: fix the problem that specify --sql will query twice

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve? 

In the previous implementation, using `--sql` option triggers query twice.

### What is changed and how it works?

This PR add a new implementation for `TableMeta` and `TableDataIR` to avoid query twice.

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

N/A

Related changes

N/A
 
### Release note

- No release note

### CODE REVIEW

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM.
Good integration test 👍

File: v4/export/ir.go [hunk 1]
```diff
@@ -83,3 +85,21 @@ type Logger interface {
...
+	}
+	nms, err := rows.Columns()
+	if err != nil {
+		return nil, err
+	}
+	for i := range nms {
+		nms[i] = wrapBackTicks(nms[i])
+	}
+	return &tableMeta{
+		colTypes:      tps,
```
[the remark this answers is not in the document]
Comment (tangenta) [reply v4/export/ir.go:102]: Updated.

File: tests/basic/run.sh [hunk 2]
```diff
@@ -31,4 +32,16 @@ run_dumpling --where "a >= 3 and a <= 9" -f "$DB_NAME.$TABLE_NAME"
...
+export DUMPLING_TEST_PORT=4000
+# Test for --sql option.
+run_sql "drop database if exists \`$DB_NAME\`;"
+run_sql "create database \`$DB_NAME\`;"
+run_sql "create sequence \`$DB_NAME\`.\`SEQUENCE_NAME\` increment by 1;"
+run_dumpling --sql "select nextval(\`$DB_NAME\`.\`SEQUENCE_NAME\`)"
+
+actual=$(grep -w "(.*)[,|;]" ${DUMPLING_OUTPUT_DIR}/result.000000000.sql | cut -c2-2)
+echo "expected 1, actual ${actual}"
+[ "$actual" = 1 ]
```
Comment (kennytm) [inline tests/basic/run.sh:47]: please also `run_sql 'select nextval()'` and check the result is 2.

**Review by kennytm — COMMENTED** [review]
rest lgtm

File: tests/basic/run.sh (hunk 2 above)
Comment (tangenta) [reply tests/basic/run.sh:48]: Updated.

**Review by kennytm — APPROVED** [review]

**Comment by kennytm** [comment]
(retry CI)

**Pull request merged**